src/boot/store.js

```diff
@@ -228,6 +234,20 @@ function listMiddleware() {
       // and ours:
       //   https://github.com/zulip/zulip-mobile/blob/master/docs/howto/debugging.md#redux-logger
       createLogger({
+        stateTransformer: state => {
+          // Beware: errors thrown in here will silently drop actions.
+          const newState = {};
+          Object.keys(state).forEach(key => {
+            if (Immutable.isImmutable(state[key])) {
+              // Make things like `state.narrows` appear as plain
+              // JS objects in the logs (much easier to read)
```

Comment:
Neat.

Have you tried using this implementation? https://github.com/andrewdavey/immutable-devtools It's a little more code, and specifically aims at the Chrome Dev Tools API... and the result seems somewhat cleaner than is possible to get by translating to plain objects-as-maps and arrays. (One difference that seems fundamental is that if we start using some map keys that aren't strings, those can't translate properly to an object-as-map.)

Ran across that in looking at the Immutable.js wiki: https://github.com/immutable-js/immutable-js/wiki/Modules-that-work-with-Immutable.js

Reply:
Ooh, that looks helpful. Another reason to prefer something like that (as noted in a code comment): the current implementation will silently drop actions if there's an error thrown from the code that makes plain objects out of `Immutable` objects.